### PR TITLE
Disable Queue

### LINKED ISSUE #734: [Queue Refactor] Add front end to disable/hide a queue forever

### PR DESCRIPTION
**Pull request by VKong6019**
# Description

Adds the frontend functionality to disable queue along with a warning modal for confirmation.

Closes #734

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] This requires a run of `yarn install`

# How Has This Been Tested?

Please describe how you tested this PR (both manually and with tests)
Provide instructions so we can reproduce. 

- [x] manual testing 😎 
- [x] aka i looked at the psql tables 

disable-queue

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code where needed 
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by VKong6019** [comment]
blocked by https://github.com/sandboxnu/office-hours/pull/757, that pr has the disable queue endpoint

File: packages/app/components/Queue/QueueListSharedComponents.tsx
Comment (IrisLiu-00) [inline packages/app/components/Queue/QueueListSharedComponents.tsx:192]: This button gets bumped up right under the TA status cards on my smaller laptop screen - any way you guys could get them to space out more? (looks like setting the InfoColumnContainer to have a flex column display + giving the TAStatuses a bottom margin should work)

File: packages/app/components/Queue/QueueListSharedComponents.tsx
Comment (IrisLiu-00) [inline packages/app/components/Queue/QueueListSharedComponents.tsx:133]: Let's add another sentence to this modal to say something like, "This queue will no longer appear in the app, and any students currently in the queue will be removed." so users know what happens upon disabling.

File: packages/app/components/Queue/QueueListSharedComponents.tsx
Comment (IrisLiu-00) [inline packages/app/components/Queue/QueueListSharedComponents.tsx:117]: This prop should probably be `isTaOrProf` or `isStaff` - we want profs to be able to disable as well.

File: packages/server/src/queue/queue.controller.ts
Comment (IrisLiu-00) [inline packages/server/src/queue/queue.controller.ts:178]: I don't think you need to check the role here because this endpoint already has role guards on it; also I dont think you need to call save() because that's being done at the end of this func already.

**Review by IrisLiu-00 — CHANGES_REQUESTED** [review]
Looks good! I just have a few nits -

File: packages/app/components/Queue/QueueListSharedComponents.tsx
Comment (VKong6019) [reply packages/app/components/Queue/QueueListSharedComponents.tsx:192]: like this?
image

**Review by IrisLiu-00 — APPROVED** [review]
Looks good! ideally we want 2 approvals per PR... so you can start bothering people about it but if its taking a long time then I would say just merge it.

**Review by nicoledanuwidjaja — APPROVED** [review]
Nice frontend!

**Pull request merged**
